src/registrar/models/domain.py

```diff
@@ -267,25 +373,234 @@ def administrative_contact(self) -> PublicContact:
 
     @administrative_contact.setter  # type: ignore
     def administrative_contact(self, contact: PublicContact):
-        # call CreateContact, if contact doesn't exist yet for domain
-        # call UpdateDomain with contact,
-        #  type options are[admin, billing, tech, security]
-        # use admin as type parameter for this contact
-        raise NotImplementedError()
+        logger.info("making admin contact")
+        if contact.contact_type != contact.ContactTypeChoices.ADMINISTRATIVE:
+            raise ValueError(
+                "Cannot set a registrant contact with a different contact type"
+            )
+        self._make_contact_in_registry(contact=contact)
+        self._update_domain_with_contact(contact, rem=False)
 
-    @Cache
-    def security_contact(self) -> PublicContact:
-        """Get or set the security contact for this domain."""
-        # TODO: replace this with a real implementation
+    def get_default_security_contact(self):
+        logger.info("getting default sec contact")
         contact = PublicContact.get_default_security()
         contact.domain = self
-        contact.email = "mayor@igorville.gov"
         return contact
 
+    def _update_epp_contact(self, contact: PublicContact):
+        """Sends UpdateContact to update the actual contact object,
+        domain object remains unaffected
+        should be used when changing email address
+        or other contact info on an existing domain
+        """
+        updateContact = commands.UpdateContact(
+            id=contact.registry_id,
+            # type: ignore
+            postal_info=self._make_epp_contact_postal_info(contact=contact),
+            email=contact.email,
+            voice=contact.voice,
+            fax=contact.fax,
+        )  # type: ignore
+
+        try:
+            registry.send(updateContact, cleaned=True)
+        except RegistryError as e:
+            logger.error(
+                "Error updating contact, code was %s error was %s" % (e.code, e)
+            )
+            # TODO - ticket 433 human readable error handling here
+
+    def _update_domain_with_contact(self, contact: PublicContact, rem=False):
+        """adds or removes a contact from a domain"""
+        logger.info(
+            "_update_domain_with_contact() received type %s " % contact.contact_type
+        )
+        domainContact = epp.DomainContact(
+            contact=contact.registry_id, type=contact.contact_type
+        )
+
+        updateDomain = commands.UpdateDomain(name=self.name, add=[domainContact])
+        if rem:
+            updateDomain = commands.UpdateDomain(name=self.name, rem=[domainContact])
```

Comment:
```suggestion
        # May be invalid syntax but I think you can do this...
        updateDomain: commands.UpdateDomain
        if rem:
            updateDomain = commands.UpdateDomain(name=self.name, rem=[domainContact])
        else:
            updateDomain = commands.UpdateDomain(name=self.name, add=[domainContact])
```
If rem=True, wouldn't this add the contact then immediately delete it? Seems potentially inefficient if not intentional

Reply:
No the command isn't sent yet, it is just making a command and avoiding an else block basically